src/frontends/tensorflow_common/src/op/adjust_saturation.cpp

```diff
@@ -0,0 +1,192 @@
+#include "openvino/op/clamp.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+// Image format conversion based on https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/image/adjust_saturation_op.cc
+
+// shared_ptr<Node> convert_rgb_to_hsv(shared_ptr<Node> images) {
+shared_ptr<tuple<shared_ptr<Node>, shared_ptr<Node>, shared_ptr<Node>>> convert_rgb_to_hsv(shared_ptr<Node> images, element::Type type) {
+
+    auto zero = make_shared<v0::Constant>(type, Shape{}, 0.0f);
+    auto one = make_shared<v0::Constant>(type, Shape{}, 1.0f);
+
+    // Find max and min across channel axis. Max = Value (V)
+    auto max_rgb = make_shared<v1::ReduceMax>(images, make_shared<v0::Constant>(element::i64, Shape{1}, vector<int64_t>{-1}), true);
+    auto min_rgb = make_shared<v1::ReduceMin>(images, make_shared<v0::Constant>(element::i64, Shape{1}, vector<int64_t>{-1}), true);
+    auto range = make_shared<v1::Subtract>(max_rgb, min_rgb);
+    auto vv = max_rgb;
+
+    // Compute Saturation (S)
+    auto ss_ = make_shared<v1::Divide>(range, vv);
+    auto ss = make_shared<v1::Select>(make_shared<v1::Greater>(vv, zero), ss_, zero);
+
+    // Compute normalization factor (for Hue calculation)
+    auto norm = make_shared<v1::Divide>(
+        make_shared<v0::Constant>(type, range->get_shape(), vector<float>{1.0f}),
+        make_shared<v1::Multiply>(
+            make_shared<v0::Constant>(type, range->get_shape(), vector<float>{6.0f}),
+            range
+        )
+    );
+
+    // Split the image tensor into R, G, B channels
+    int batch_dim = images->get_shape().size() - 1;
+    auto channels = make_shared<v1::Split>(images, make_shared<v0::Constant>(element::i64, Shape{}, batch_dim), 3);
+    auto r = channels->output(0);
+    auto g = channels->output(1);
+    auto b = channels->output(2);
+
+    // Compute Hue (H)
+    
+    // Determine which component is the max (V) to compute Hue (H)
+    auto r_eq_v = make_shared<v1::Equal>(r, vv);
+    auto g_eq_v = make_shared<v1::Equal>(g, vv);
+    // auto b_eq_v = make_shared<v1::Equal>(b, vv);
+
+    // r == vv: hh = norm * (g - b)
+    auto hue_case_r = make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(g, b));
+
+    // g == vv: hh = norm * (b - r) + 2.0 / 6.0
+    auto hue_case_g = make_shared<v1::Add>(
+        make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(b, r)),
+        make_shared<v0::Constant>(element::f32, Shape{}, std::vector<float>{2.0f / 6.0f})
+    );
+
+    // b == vv: hh = norm * (r - g) + 4.0 / 6.0
+    auto hue_case_b = make_shared<v1::Add>(
+        make_shared<v1::Multiply>(norm, make_shared<v1::Subtract>(r, g)),
+        make_shared<v0::Constant>(element::f32, Shape{}, std::vector<float>{4.0f / 6.0f})
+    );
+
+    // Select hue based on the maximum component
+    // Check if `r` is the max, otherwise check if `g` is the max, if not use `b`'s hue
```

Comment:
```suggestion
    // select hue based on the maximum component
    // check if `r` is the max, otherwise check if `g` is the max, if not use `b`'s hue
```